ts/src/main/scala/ch/epfl/ts/component/Component.scala

```diff
@@ -51,6 +94,24 @@ final class ComponentBuilder(val system: ActorSystem) {
     instances = new ComponentRef(system.actorOf(props, name), props.clazz, name, this) :: instances
     instances.head
   }
+  
+  /**
+   * Gracefully stop all managed components.
+   * When all stops are successful, we clear the `instances` list.
+   * 
+   * @return A future which completes when *all* managed actors have shut down.
+   */
+  def shutdownManagedActors(timeout: FiniteDuration = 3 seconds): Future[Unit] = {
```

Comment:
what's the relationship between `stop` and `shutdownManagedActors`? We need both, or we can just keep one?